In [2]:
import pandas as pd
import numpy as np

In [4]:
df = pd.read_csv('task1_dataset.csv')

In [6]:
df.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)                 1   
1                          Japanese  ...  Botswana Pula(P)                 1   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)                 1   
3                   Japanese, Sushi  ...  Botswana Pula(P)                 0   
4                  Japanese, Korean  ...  Botswana Pula(P)                 1   

   Has Online delivery  Is delivering now  Switch to order menu Price range  \
0                    0                  0                    No           3   
1                    0                  0                    No           3   
2                    0                  0                    No           4   
3                    0                  0                    No           4   
4                    0                  0                    No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

##  Step 1: Explore Statistical Summary
         Goal: Get a quick overview of the dataset's numerical and categorical features.

In [9]:
# View statistical summary for numeric columns
df.describe()

Restaurant ID  Country Code    Longitude     Latitude  \
count   9.551000e+03   9551.000000  9551.000000  9551.000000   
mean    9.051128e+06     18.365616    68.176688    27.274530   
std     8.791521e+06     56.750546    38.712304     9.247629   
min     5.300000e+01      1.000000  -157.948486   -41.330428   
25%     3.019625e+05      1.000000    77.094647    28.501221   
50%     6.004089e+06      1.000000    77.198219    28.574530   
75%     1.835229e+07      1.000000    77.289188    28.642989   
max     1.850065e+07    216.000000   174.832089    55.976980   

       Average Cost for two  Has Table booking  Has Online delivery  \
count           9551.000000        9551.000000          9551.000000   
mean            1199.210763           0.121244             0.256622   
std            16121.183073           0.326428             0.436792   
min                0.000000           0.000000             0.000000   
25%              250.000000           0.000000             0.000000   
50%              400.000000           0.000000             0.000000   
75%              700.000000           0.000000             1.000000   
max           800000.000000           1.000000             1.000000   

       Is delivering now  Price range  Aggregate rating         Votes  
count        9551.000000  9551.000000       9551.000000   9551.000000  
mean            0.003560     1.804837          2.666370    156.909748  
std             0.059561     0.905609          1.516378    430.169145  
min             0.000000     1.000000          0.000000      0.000000  
25%             0.000000     1.000000          2.500000      5.000000  
50%             0.000000     2.000000          3.200000     31.000000  
75%             0.000000     2.000000          3.700000    131.000000  
max             1.000000     4.000000          4.900000  10934.000000

#### Analyze Outliers in Key Columns
    Identify Outliers in "Average Cost for Two"


In [12]:
# Restaurants with the highest cost
df[df['Average Cost for two'] == df['Average Cost for two'].max()]


Restaurant ID           Restaurant Name  Country Code     City  \
9283        7402935                      Skye            94  Jakarta   
9284        7410290  Satoo - Hotel Shangri-La            94  Jakarta   

                                                Address  \
9283  Menara BCA, Lantai 56, Jl. MH. Thamrin, Thamri...   
9284               Hotel Shangri-La, Jl. Jend. Sudirman   

                           Locality                        Locality Verbose  \
9283  Grand Indonesia Mall, Thamrin  Grand Indonesia Mall, Thamrin, Jakarta   
9284     Hotel Shangri-La, Sudirman     Hotel Shangri-La, Sudirman, Jakarta   

       Longitude  Latitude                    Cuisines  ...  \
9283  106.821999 -6.196778        Italian, Continental  ...   
9284  106.818961 -6.203292  Asian, Indonesian, Western  ...   

                    Currency Has Table booking  Has Online delivery  \
9283  Indonesian Rupiah(IDR)                 0                    0   
9284  Indonesian Rupiah(IDR)                 0                    0   

      Is delivering now  Switch to order menu Price range  Aggregate rating  \
9283                  0                    No           3               4.1   
9284                  0                    No           3               4.6   

      Rating color Rating text Votes  
9283         Green   Very Good  1498  
9284    Dark Green   Excellent   873  

[2 rows x 21 columns]

In [14]:
# Restaurants with the lowest cost (possibly errors)
df[df['Average Cost for two'] == df['Average Cost for two'].min()]

Restaurant ID                 Restaurant Name  Country Code       City  \
84         17284105                   Cookie Shoppe           216     Albany   
85         17284302   El Vaquero Mexican Restaurant           216     Albany   
87         17284211  Pearly's Famous Country Cookng           216     Albany   
201        17330155                        Deorio's           216   Columbus   
240        17334679                          Azteca           216  Davenport   
277        17342548   Happy Joe's Pizza & Ice Cream           216    Dubuque   
346        17606621            HI Lite Bar & Lounge           216     Miller   
397        17582499                     Royal Hotel           216  Pocatello   
407        17582627                   Senor Iguanas           216  Pocatello   
634         3400073     The Latitude - Radisson Blu             1       Agra   
637         3400346                 Sheroes Hangout             1       Agra   
639         3400391                  Chapter 1 Cafe             1       Agra   
677        18317988                  The BrewMaster             1  Allahabad   
851        18279289            BMG - All Day Dining             1   Dehradun   
2364        2300497   Atmosphere Grill Cafe Sheesha             1     Kanpur   
2368       18312106                      UrbanCrave             1     Kanpur   
9242        3900245              Deena Chat Bhandar             1   Varanasi   
9254       18246202                 VNS Live Studio             1   Varanasi   

                                                Address  \
84                   115 N Jackson St, Albany, GA 31701   
85                     2700 Dawson Rd, Albany, GA 31707   
87                 814 N Slappey Blvd, Albany, GA 31701   
201           3201 Macon Rd Ste 167, Columbus, GA 31906   
240          4811 N Brady St Ste 3, Davenport, IA 52806   
277                   855 Century Dr, Dubuque, IA 52002   
346                109 N Broadway Ave, Miller, SD 57362   
397            11 E Main St, Lava Hot Springs, ID 83246   
407                  961 Hiline Rd, Pocatello, ID 83201   
634     Radisson Blu, Taj East Gate Road, Tajganj, Agra   
637   Opposite The Gateway Hotel, Fatehabad Road, Ta...   
639   1374 K/1375 K, 2nd floor, Dinesh Nagar, Fatehb...   
677        Near Vishal Megamart, Civil Lines, Allahabad   
851                140 A, Rajpur Road, Jakhan, Dehradun   
2364  8th Floor, J.S. Tower, 16/106 - Mall Road, Kan...   
2368  14/125, The Mall, Mall Road, Colonelganj, Para...   
9242    D-47/184, Luxa Road, Dashaswmedh Road, Varanasi   
9254  Hotel Varuna Ground Floor, 22 Gulab Bagh, Sigr...   

                   Locality             Locality Verbose   Longitude  \
84                   Albany               Albany, Albany  -84.154000   
85                   Albany               Albany, Albany  -84.219400   
87                   Albany               Albany, Albany  -84.175900   
201                Columbus           Columbus, Columbus  -84.942700   
240               Davenport         Davenport, Davenport  -90.568300   
277                 Dubuque             Dubuque, Dubuque  -90.726400   
346                  Miller               Miller, Miller  -98.989100   
397        Lava Hot Springs  Lava Hot Springs, Pocatello -112.012700   
407               Pocatello         Pocatello, Pocatello -112.443300   
634   Radisson Blu, Tajganj  Radisson Blu, Tajganj, Agra   78.057044   
637                 Tajganj                Tajganj, Agra   78.040165   
639                 Tajganj                Tajganj, Agra   78.032615   
677             Civil Lines       Civil Lines, Allahabad   81.832796   
851                  Jakhan             Jakhan, Dehradun   78.068890   
2364              Mall Road            Mall Road, Kanpur   80.354002   
2368                 Parade               Parade, Kanpur   80.342796   
9242       Dashaswmedh Road   Dashaswmedh Road, Varanasi   82.992199   
9254                  Sigra              Sigra, Varanasi   82.99169

2. Analyze Restaurants with the Most Votes
   
Let’s find which restaurants have the highest customer engagement.

👉 Find restaurants with the most votes:



In [17]:
# Restaurants with the highest votes
df[df['Votes'] == df['Votes'].max()]

Restaurant ID Restaurant Name  Country Code       City  \
728          51705            Toit             1  Bangalore   

                                               Address     Locality  \
728  298, Namma Metro Pillar 62, 100 Feet Road, Ind...  Indiranagar   

           Locality Verbose  Longitude   Latitude                  Cuisines  \
728  Indiranagar, Bangalore  77.640709  12.979166  Italian, American, Pizza   

     ...            Currency Has Table booking  Has Online delivery  \
728  ...  Indian Rupees(Rs.)                 0                    0   

     Is delivering now  Switch to order menu Price range  Aggregate rating  \
728                  0                    No           4               4.8   

     Rating color Rating text  Votes  
728    Dark Green   Excellent  10934  

[1 rows x 21 columns]

3. Explore Restaurants with Perfect or No Ratings

Let’s check for top-rated and unrated restaurants.

👉 Find top-rated restaurants (4.9 rating):

In [20]:
# Restaurants with the highest rating (4.9)
df[df['Aggregate rating'] == df['Aggregate rating'].max()]


Restaurant ID                           Restaurant Name  Country Code  \
3           6318506                                      Ooma           162   
8           6300010  Spiral - Sofitel Philippine Plaza Manila           162   
10          6309903                          Silantro Fil-Mex           162   
39          6600427                                Coco Bambu            30   
48          7300955                        Braseiro da G��vea            30   
...             ...                                       ...           ...   
9484          75989            Restaurant Mosaic @ The Orient           189   
9514        5800176                          Ministry of Crab           191   
9524        6004011                              Gaga Manjero           208   
9538        5901782                                 Starbucks           208   
9540        5927248                          Draft Gastro Pub           208   

                  City                                            Address  \
3     Mandaluyong City  Third Floor, Mega Fashion Hall, SM Megamall, O...   
8           Pasay City  Plaza Level, Sofitel Philippine Plaza Manila, ...   
10          Pasig City       75 East Capitol Drive, Kapitolyo, Pasig City   
39           Bras�_lia  SCES, Trecho 2, Conjunto 13/36, Setor de Clube...   
48      Rio de Janeiro  Pra�_a Santos Dumont, 116, G��vea, Rio de Janeiro   
...                ...                                                ...   
9484          Pretoria  The Orient Boutique Hotel, Crocodile River Val...   
9514           Colombo       Old Colombo Dutch Hospital, Fort, Colombo 01   
9524            Ankara  Gazi Osman Pa��a Mahallesi, Filistin Caddesi, ...   
9538         ��stanbul  Bebek Mahallesi, Cevdetpa��a Caddesi, No 30/A,...   
9540         ��stanbul  Caddebostan Mahallesi, Ba��dat Caddesi, No 349...   

                                         Locality  \
3          SM Megamall, Ortigas, Mandaluyong City   
8     Sofitel Philippine Plaza Manila, Pasay City   
10                                      Kapitolyo   
39                 Setor De Clubes Esportivos Sul   
48                                         G��vea   
...                                           ...   
9484                                    West Park   
9514                     Old Dutch Hospital, Fort   
9524                             Gazi Osman Pa��a   
9538                                        Bebek   
9540                                  Caddebostan   

                                       Locality Verbose   Longitude  \
3     SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475   
8     Sofitel Philippine Plaza Manila, Pasay City, P...  120.980090   
10                                Kapitolyo, Pasig City  121.057916   
39            Setor De Clubes Esportivos Sul, Bras�_lia  -47.868500   
48                               G��vea, Rio de Janeiro  -43.227042   
...                                                 ...         ...   
9484                                West Park, Pretoria   27.999097   
9514                  Old Dutch Hospital, Fort, Colombo   79.844239   
9524                           Gazi Osman Pa��a, Ankara   32.869800   
9538                                   Bebek, ��stanbul   29.043734   
9540                             Caddebostan, ��stanbul   29.074116   

       Latitude                 Cuisines  ...               Currency  \
3     14.585318          Japanese, Sushi  ...       Botswana Pula(P)   
8     14.552990  European, Asian, Indian  ...       Botswana Pula(P)   
10    14.567689        Filipino, Mexican  ...       Botswana Pula(P)   
39   -15.819000            International  ...     Brazilian Real(R$)   
48   -22.973507      Brazilian, Bar Food  ...     Brazilian Real(R$)   
...         ...                      ...  ...                    ...   
9484 -25.761238                   French  ...                Rand(R)   
9514   6.932814                  Seafood  ...  Sri Lankan Rupee(LK

In [22]:
# Restaurants with no rating
df[df['Aggregate rating'] == 0.0]


Restaurant ID         Restaurant Name  Country Code            City  \
30          6600060          Sandubas Caf��            30       Bras�_lia   
58          7305048   Quiosque Chopp Brahma            30  Rio de Janeiro   
69          6710645        Cantinho da Gula            30      S��o Paulo   
77          6701419           Divino Fog��o            30      S��o Paulo   
78          6703956             Super Grill            30      S��o Paulo   
...             ...                     ...           ...             ...   
9109       18277023           Bread & Pasta             1           Noida   
9110       18313203           Chillies Cafe             1           Noida   
9111       18254559                Platters             1           Noida   
9112       18255132               The Grand             1           Noida   
9351       18273002  Damascena Coffee House           215      Birmingham   

                                                Address  \
30    Edif�_cio Jos�� Severo, SCS 6, Bloco A, Loja 9...   
58    Madureira Shopping - Loja 289/290, Piso 2, Est...   
69    Rua Pedroso Alvarenga, 522, Itaim Bibi, S��o P...   
77    Shopping Metr�� Santa Cruz - Piso L2, Rua Domi...   
78    Shopping Morumbi - Piso Lazer, Avenida Roque P...   
...                                                 ...   
9109  Shop 15, Near HDFC Bank, Tulip Mall, Sector 48...   
9110                 G-11, Tulip Mall, Sector 48, Noida   
9111  Shop 1, Tulip Mall, Near HDFC Bank, Sector 48,...   
9112                  G-5, Tulip Mall, Sector 48, Noida   
9351             133 Alcester Road, Moseley, Birmingham   

                                      Locality  \
30                                     Asa Sul   
58                                   Madureira   
69                                  Itaim Bibi   
77    Shopping Metr�� Santa Cruz, Vila Mariana   
78               Shopping Morumbi, Santo Amaro   
...                                        ...   
9109              Tulip Mall, Sector 48, Noida   
9110              Tulip Mall, Sector 48, Noida   
9111              Tulip Mall, Sector 48, Noida   
9112              Tulip Mall, Sector 48, Noida   
9351                                   Moseley   

                                       Locality Verbose  Longitude   Latitude  \
30                                   Asa Sul, Bras�_lia -47.890167 -15.797000   
58                            Madureira, Rio de Janeiro -43.341164 -22.870413   
69                               Itaim Bibi, S��o Paulo -46.675667 -23.581000   
77    Shopping Metr�� Santa Cruz, Vila Mariana, S��o... -46.637167 -23.599500   
78            Shopping Morumbi, Santo Amaro, S��o Paulo -46.698574 -23.622925   
...                                                 ...        ...        ...   
9109                Tulip Mall, Sector 48, Noida, Noida  77.367303  28.557929   
9110                Tulip Mall, Sector 48, Noida, Noida  77.367188  28.557842   
9111                Tulip Mall, Sector 48, Noida, Noida  77.367322  28.557930   
9112                Tulip Mall, Sector 48, Noida, Noida  77.367188  28.557852   
9351                                Moseley, Birmingham  -1.888555  52.446302   

                                  Cuisines  ...            Currency  \
30                         Brazilian, Cafe  ...  Brazilian Real(R$)   
58                     Bar Food, Brazilian  ...  Brazilian Real(R$)   
69                               Brazilian  ...  Brazilian Real(R$)   
77                      Brazilian, Mineira  ...  Brazilian Real(R$)   
78                               Brazilian  ...  Brazilian Real(R$)   
...                                    ...  ...                 ...   
9109                             Fast Food  ...  Indian Rupees(Rs.)   
9110                             Fast Food  ...  Indian Rupees(Rs.)   
9111                 North Indian, Chinese  ...  Indian Rupees(Rs.)   
9112                  Raw Meats, Fast Food  ...  Indian Rupees(Rs.)   
9351  Greek, Medi

 Step 1: Identify the Top Cities by Restaurant Count

Objective: Find out which cities have the most and least restaurants.

In [25]:
# Top 10 cities with the most restaurants
df['City'].value_counts().head(10)

City
New Delhi       5473
Gurgaon         1118
Noida           1080
Faridabad        251
Ghaziabad         25
Bhubaneshwar      21
Amritsar          21
Ahmedabad         21
Lucknow           21
Guwahati          21
Name: count, dtype: int64

Step 2: Analyze Rated vs. Not Rated Restaurants in Each City

Objective: Understand how many restaurants are rated vs. unrated across cities.

👉 Check the rated/unrated distribution by city:

In [28]:
# Count rated and unrated restaurants by city
df.groupby('City')['Aggregate rating'].value_counts()


City       Aggregate rating
Abu Dhabi  4.0                 5
           4.6                 4
           4.2                 2
           4.3                 2
           4.4                 2
                              ..
��stanbul  4.7                 2
           4.9                 2
           4.1                 1
           4.3                 1
           4.5                 1
Name: count, Length: 946, dtype: int64

Step 3: Find Cities with the Highest Average Rating

Objective: Discover which cities have the best-rated restaurants on average.

👉 Find the top-rated cities:

In [31]:
# Average rating by city (sorted)
print(df.groupby('City')['Aggregate rating'].mean().sort_values(ascending=False).head(10))


City
Inner City          4.900000
Quezon City         4.800000
Makati City         4.650000
Pasig City          4.633333
Mandaluyong City    4.625000
Beechworth          4.600000
London              4.535000
Taguig City         4.525000
Tagaytay City       4.500000
Secunderabad        4.500000
Name: Aggregate rating, dtype: float64


### 📊 City-Wise Analysis Summary:

1. **Top Cities by Restaurant Count:**
   - **New Delhi** leads with **5473** restaurants—significantly more than any other city.  
   - Other major cities include **Gurgaon (1118)** and **Noida (1080)**.

2. **Rated vs. Unrated Restaurants:**
   - Cities like **Abu Dhabi** and **Istanbul** show a variety of ratings.  
   - We'll explore **cities with high unrated** entries next.

3. **Cities with the Highest Average Ratings:**
   - **Inner City** (4.9) and **Quezon City** (4.8) have the **highest** average ratings.  
   - **Makati City** and **Pasig City** also have consistently **high ratings**.

4. **Next Steps (Tomorrow, InshaAllah):**
   - Continue exploring **city-level trends**.
   - Analyze **customer engagement** and **vote patterns** by city.


Next Steps for City-Wise Analysis:

   1️⃣ Find Cities with the Most Customer Engagement (Votes)
   
   👉 Run this to see which cities have the most votes:

In [35]:
# Top 10 cities with the most restaurants
print(df['City'].value_counts().head(10))


City
New Delhi       5473
Gurgaon         1118
Noida           1080
Faridabad        251
Ghaziabad         25
Bhubaneshwar      21
Amritsar          21
Ahmedabad         21
Lucknow           21
Guwahati          21
Name: count, dtype: int64


📊 Step 2: Find Cities with the Most Unrated Restaurants

👉 Check how many unrated (0.0 rating) restaurants exist per city:

In [38]:
# Count of Not Rated (0.0) restaurants per city
df[df['Aggregate rating'] == 0.0]['City'].value_counts().head(10)


City
New Delhi         1425
Noida              384
Gurgaon            228
Faridabad          100
S��o Paulo           3
Davenport            2
Ghaziabad            2
Bras�_lia            1
Rio de Janeiro       1
Pocatello            1
Name: count, dtype: int64

Step 3: Compare High-Rated vs. Low-Rated Cities

👉 Find cities with the most highly rated (4.5+) restaurants:

In [41]:
# Count of high-rated restaurants (4.5+) per city
print(df[df['Aggregate rating'] >= 4.5]['City'].value_counts().head(10))


City
New Delhi         28
London            15
Gurgaon           12
Rio de Janeiro    11
Tampa Bay         10
Auckland           9
Bangalore          9
Dubai              9
Rest of Hawaii     9
Orlando            9
Name: count, dtype: int64


In [43]:
# Count of low-rated restaurants (<2.5) per city
print(df[df['Aggregate rating'] < 2.5]['City'].value_counts().head(10))


City
New Delhi     1522
Noida          429
Gurgaon        262
Faridabad      102
Ghaziabad        3
S��o Paulo       3
Davenport        2
Birmingham       1
Auckland         1
Nagpur           1
Name: count, dtype: int64


## 📊 Key Insights from City-Wise Analysis

### 1️⃣ Cities with the Most Restaurants
- **New Delhi (5473), Gurgaon (1118), and Noida (1080)** dominate the dataset.  
- These cities have a **high restaurant density**, making them major food hubs.

### 2️⃣ Cities with the Most Unrated (0.0) Restaurants
- **New Delhi (1425), Noida (384), Gurgaon (228)** have the highest number of **unrated** restaurants.  
- Indicates **low customer engagement or missing reviews** in these areas.

### 3️⃣ Cities with the Most High-Rated (4.5+) Restaurants
- **New Delhi has 28** highly-rated restaurants.  
- Other top-rated cities include **Gurgaon (12), Rio de Janeiro (11), and Tampa Bay (10)**.

### 4️⃣ Cities with the Most Poorly Rated (<2.5) Restaurants
- **New Delhi (1522), Noida (429), Gurgaon (262)** have the highest number of **poorly rated** restaurants.  
- **London (15) has the most low-rated restaurants internationally.**  
- **Bangalore, Dubai, and Auckland** also have a significant number of **low-rated** restaurants.

## 🔍 Key Takeaways:
- **New Delhi dominates** in both **restaurant count** and **low-rated/unrated** businesses.  
- **High-rated restaurants are fewer**, meaning **only a few truly excellent** places exist.  
- **Cities like Rio de Janeiro, Tampa Bay, and Auckland** have a **higher percentage of highly-rated restaurants**, indicating **better quality control**.



## 2️⃣ Cuisine Analysis

Step 1: Identify the Most Popular Cuisines

Objective: Find out which cuisines appear most frequently in the dataset.

In [51]:
# Top 10 most popular cuisines
print(df['Cuisines'].value_counts().head(10))


Cuisines
North Indian                      936
North Indian, Chinese             511
Chinese                           354
Fast Food                         354
North Indian, Mughlai             334
Cafe                              299
Bakery                            218
North Indian, Mughlai, Chinese    197
Bakery, Desserts                  170
Street Food                       149
Name: count, dtype: int64


📌 Step 2: Find Which Cuisines Have the Highest Ratings

Objective: Identify which cuisines get higher customer satisfaction (ratings).

In [55]:
# Average rating by cuisine (sorted)
print(df.groupby('Cuisines')['Aggregate rating'].mean().sort_values(ascending=False).head(10))


Cuisines
Italian, Deli               4.9
Hawaiian, Seafood           4.9
American, Sandwich, Tea     4.9
Continental, Indian         4.9
European, Asian, Indian     4.9
European, Contemporary      4.9
European, German            4.9
BBQ, Breakfast, Southern    4.9
American, Coffee and Tea    4.9
Sunda, Indonesian           4.9
Name: Aggregate rating, dtype: float64


📌 Step 3: Compare Local vs. International Cuisines

Objective: See if customers prefer local cuisines over international ones.

In [62]:
# Fill NaN values with an empty string before searching
df['Cuisines'] = df['Cuisines'].fillna('')

# Count restaurants serving international vs. local cuisines
print(df[df['Cuisines'].str.contains("Italian|Chinese|Mexican|American")].shape[0])  # Example for international cuisines
print(df[df['Cuisines'].str.contains("Indian|Filipino|Brazilian")].shape[0])  # Example for local cuisines


3670
4296


Find the Most Common Cuisines Overall

In [65]:
# Top 10 most popular cuisines
print(df['Cuisines'].value_counts().head(10))

Cuisines
North Indian                      936
North Indian, Chinese             511
Chinese                           354
Fast Food                         354
North Indian, Mughlai             334
Cafe                              299
Bakery                            218
North Indian, Mughlai, Chinese    197
Bakery, Desserts                  170
Street Food                       149
Name: count, dtype: int64


 Analyze Which Cuisines Have the Highest Ratings

In [74]:
# Average rating by cuisine (sorted)
print(df.groupby('Cuisines')['Aggregate rating'].mean().sort_values(ascending=False).head(10))


Cuisines
American, Caribbean, Seafood       4.9
American, BBQ, Sandwich            4.9
American, Coffee and Tea           4.9
American, Burger, Grill            4.9
Italian, Deli                      4.9
Mughlai, Lucknowi                  4.9
European, German                   4.9
European, Contemporary             4.9
European, Asian, Indian            4.9
Mexican, American, Healthy Food    4.9
Name: Aggregate rating, dtype: float64


📊 Key Insights from Cuisine Analysis

🍽️ Most Popular Cuisines (Top 10)

1️⃣ North Indian (936) – The most widely available cuisine.

2️⃣ North Indian, Chinese (511) – Fusion of Indian & Chinese is common.

3️⃣ Chinese (354) & Fast Food (354) – Equally popular.

4️⃣ Cafe (299) & Bakery (218) – Many restaurants serve these options.
    
5️⃣ Street Food (149) – Indicates strong demand for local snacks.

📌 Takeaway:

Indian cuisine dominates the dataset.
    
Fusion cuisines (Indian + Chinese, Indian + Mughlai) are common.
    
Fast Food & Bakery items are highly available across cities.
    
⭐ Highest Rated Cuisines (4.9 Rating)
    
1️⃣ American Cuisine (BBQ, Seafood, Coffee, Grill) – Consistently top-rated.
    
2️⃣ European Cuisine (German, Contemporary, Fusion with Asian & Indian) – Well-liked.
    
3️⃣ Mughlai, Lucknowi – One of the highest-rated Indian cuisines.
    
4️⃣ Mexican-American Healthy Food – Health-conscious dining is appreciated.

📌 Takeaway:

Fusion cuisines perform exceptionally well.
    
Customers love premium, Western-style meals.
    
Traditional Mughlai food is highly rated among Indian cuisines.
    

In [71]:
# Find the average rating for each price range
print(df.groupby('Price range')['Aggregate rating'].mean())

Price range
1    1.999887
2    2.941054
3    3.683381
4    3.817918
Name: Aggregate rating, dtype: float64


📊 Key Insights from Price vs. Rating Analysis

📌 Price Ranges and Their Average Ratings:

1️⃣ Price Range 1 (Lowest): 1.99 ⭐

Budget-friendly restaurants have the lowest average ratings.

Likely due to lower quality, service, or food variety.

2️⃣ Price Range 2 (Affordable): 2.94 ⭐

Moderate improvement in ratings.

Some budget restaurants may offer better quality.

3️⃣ Price Range 3 (Mid-Range): 3.68 ⭐

Noticeable jump in customer satisfaction.

Suggests that mid-range restaurants offer better quality.

4️⃣ Price Range 4 (High-End): 3.81 ⭐

The highest-rated category.

Customers generally expect and receive better service & food quality in expensive places.

> Compare Ratings of Restaurants with vs. without These Services

In [81]:
# Average rating for restaurants with and without table booking
print(df.groupby('Has Table booking')['Aggregate rating'].mean())

# Average rating for restaurants with and without online delivery
print(df.groupby('Has Online delivery')['Aggregate rating'].mean())


Has Table booking
0    2.559359
1    3.441969
Name: Aggregate rating, dtype: float64
Has Online delivery
0    2.465296
1    3.248837
Name: Aggregate rating, dtype: float64


📊 Key Insights from Service Availability Analysis

🏷️ Impact of Table Booking on Ratings:

Restaurants without Table Booking (0): 2.56 ⭐ (Lower rating)

Restaurants with Table Booking (1): 3.44 ⭐ (Higher rating)

📌 What This Means:

✅ Restaurants with table booking get better ratings because:

Customers may prefer a more organized, hassle-free experience.
    
Likely to be higher-end restaurants with better service.
    
📦 Impact of Online Delivery on Ratings:
    
Restaurants without Online Delivery (0): 2.47 ⭐ (Lower rating)
    
Restaurants with Online Delivery (1): 3.25 ⭐ (Higher rating)

📌 What This Means:

✅ Restaurants with online delivery get higher ratings because:

Customers value convenience and easy access to food.
    
More people leave reviews due to a higher order volume.

 📊 Level 1 – Task 2: Descriptive Analysis

## 📌 **Objective**
- Understand the dataset using **descriptive statistics**.
- Analyze patterns in **key columns (cost, ratings, and services)**.
- Summarize **trends and insights** from the data.

---

## 1️⃣ **City-Wise Analysis 🌍**
### 🔹 **Top Cities by Restaurant Count**
- **New Delhi (5473), Gurgaon (1118), and Noida (1080)** have the **highest** number of restaurants.  
- Indicates **high food industry concentration** in these cities.

### 🔹 **Rated vs. Unrated Restaurants by City**
- **New Delhi (1425), Noida (384), Gurgaon (228)** have the most **unrated** restaurants.  
- Suggests **low customer engagement** in certain areas.

### 🔹 **Cities with the Highest Average Ratings**
- **Inner City (4.9), Quezon City (4.8), and Makati City (4.65)** have the highest-rated restaurants.  
- **New Delhi has the most high-rated restaurants (28).**

### 🔹 **Cities with the Most Low-Rated Restaurants**
- **New Delhi (1522), Noida (429), Gurgaon (262)** have the most **poorly rated** restaurants.  
- Suggests **quality issues** in certain areas.

### ✅ **Key Takeaways**
✔ **New Delhi dominates in restaurant count, but also has many unrated/low-rated places.**  
✔ **Some smaller cities have better average ratings than major food hubs.**  

---

## 2️⃣ **Cuisine Analysis 🍽️**
### 🔹 **Most Popular Cuisines**
| Rank | Cuisine | Count |
|------|-------------------------|-------|
| 1️⃣ | North Indian | 936 |
| 2️⃣ | North Indian, Chinese | 511 |
| 3️⃣ | Chinese | 354 |
| 4️⃣ | Fast Food | 354 |
| 5️⃣ | North Indian, Mughlai | 334 |
| 6️⃣ | Cafe | 299 |
| 7️⃣ | Bakery | 218 |
| 8️⃣ | North Indian, Mughlai, Chinese | 197 |
| 9️⃣ | Bakery, Desserts | 170 |
| 🔟 | Street Food | 149 |

- **Indian cuisine dominates** the dataset.
- **Fusion cuisines (North Indian + Chinese, North Indian + Mughlai) are common.**  
- **Fast food and bakery items are widely available.**  

### 🔹 **Highest-Rated Cuisines (4.9 Rating)**
| Cuisine | Rating |
|--------------------------|-------|
| American, Caribbean, Seafood | 4.9 |
| American, BBQ, Sandwich | 4.9 |
| American, Coffee and Tea | 4.9 |
| Italian, Deli | 4.9 |
| Mughlai, Lucknowi | 4.9 |
| European, German | 4.9 |
| Mexican, American, Healthy Food | 4.9 |

### ✅ **Key Takeaways**
✔ **Fusion cuisines perform exceptionally well.**  
✔ **American, European, and premium cuisines receive top ratings.**  
✔ **Traditional Indian cuisines like Mughlai & Lucknowi are highly rated.**  

---

## 3️⃣ **Price vs. Rating Analysis 💰**
| Price Range | Average Rating |
|------------|---------------|
| **1 (Low)** | **1.99 ⭐** |
| **2 (Affordable)** | **2.94 ⭐** |
| **3 (Mid-Range)** | **3.68 ⭐** |
| **4 (High-End)** | **3.81 ⭐** |

### ✅ **Key Takeaways**
✔ **Higher-priced restaurants get better ratings.**  
✔ **Mid-range (Price 3) restaurants still perform well, offering good value for money.**  
✔ **Cheaper restaurants struggle to get high ratings.**  

---

## 4️⃣ **Service Availability Analysis 📦**
| Service | Average Rating |
|-------------------|---------------|
| **No Table Booking (0)** | **2.56 ⭐** |
| **Table Booking Available (1)** | **3.44 ⭐** |
| **No Online Delivery (0)** | **2.47 ⭐** |
| **Online Delivery Available (1)** | **3.25 ⭐** |

### ✅ **Key Takeaways**
✔ **Restaurants with table booking receive better ratings.**  
✔ **Online delivery improves customer satisfaction.**  
✔ **Customers prefer convenience services like reservations & home delivery.**  

---

## 🎯 **Final Summary**
✔ **City-Wise Analysis** – Found top cities, rating patterns, and customer engagement.  
✔ **Cuisine Analysis** – Identified popular cuisines and top-rated food categories.  
✔ **Price vs. Rating Analysis** – Proved that expensive restaurants receive higher ratings.  
✔ **Service Availability Analysis** – Confirmed that table booking & online delivery improve satisfaction.  

---

In [88]:
# Save to CSV
df.to_csv('task2_dataset.csv', index=False)